tests/Unit/ControlSystem/Test_Averager.cpp

```diff
@@ -0,0 +1,217 @@
+      // derivatives. The second derivative is only a first order approximation,
+      // which is why we enforce a less stringent check here.
+      CHECK(custom_approx(result_t[2][0]) == analytic_func[2]);
+      CHECK(custom_approx(result_f[2][0]) == analytic_func[2]);
+    }
+
+    t += dt;
+  }
+}
+
+SPECTRE_TEST_CASE("Unit.ControlSystem.Averager.SemiAnalytic",
+                  "[ControlSystem][Unit]") {
+  double t = 0.0;
+  const constexpr size_t deriv_order = 2;
+  const double final_time = 5.0;
+
+  // The equations suggests that our exponential averager is equivalent
+  // to the simple exponential averaging method of
+  // F_avg(t) = \alpha*F(t) + (1 - \alpha)*F_avg(t_{-1}}
+  // where \alpha = \tau_m / (W(t)*D) and D = 1.0 + \tau_m/tau_avg
+  // However, the exponential averager that we are testing has time varying
+  // weight, so we allow \alpha to be a function of time and update at
+  // each timestep.
+
+  // some vars for a simple exponential average to compare against
+  std::array<DataVector, deriv_order + 1> avg_values{{{0.0}, {0.0}, {0.0}}};
+  double avg_time = 0.0;
+
+  // the measurement timescale (tau_m)
+  const double tau_m = 0.1;
+  const double avg_tscale_fac = 1.0;
+  const double damping_time = 0.1;
+  const double denom = 1.0 + tau_m / (avg_tscale_fac * damping_time);
+  // the average weight, represents W(t) in the above comment
+  double avg_weight = 0.0;
+
+  Averager<deriv_order> averager(avg_tscale_fac, true);
+
+  // define custom approx for second derivative checks
+  Approx custom_approx = Approx::custom().epsilon(1.0e-12).scale(1.0);
+
+  while (t < final_time) {
+    // test using an analytic function f(t) = t**2
+    const DataVector analytic_func = {square(t), 2.0 * t, 2.0};
+
+    // update exponential averager
+    averager.update(t, {analytic_func[0]}, {damping_time});
+    // compare values once averager has sufficient data
+    if (averager.have_valid_data(t)) {
+      // update the weight and \alpha
+      avg_weight = (tau_m + avg_weight) / denom;
+      const double alpha = tau_m / avg_weight / denom;
+
+      // do simple average of time, analytic func and its analytic derivatives
+      avg_time = alpha * t + (1.0 - alpha) * avg_time;
+      avg_values[0] = alpha * analytic_func[0] + (1.0 - alpha) * avg_values[0];
+      avg_values[1] = alpha * analytic_func[1] + (1.0 - alpha) * avg_values[1];
+      avg_values[2] = alpha * analytic_func[2] + (1.0 - alpha) * avg_values[2];
+
+      auto result = averager(t);
+
+      // check that the effective times agree with the averaged time
+      CHECK(approx(averager.average_time(t)[0]) == avg_time);
+      // check function value
+      CHECK(approx(result[0][0]) == avg_values[0][0]);
+      // check first derivative
+      CHECK(approx(result[1][0]) == avg_values[1][0]);
+      // check second derivative
+      // Again, this check is slightly looser than the others due to
+      // numerical differentiation (see comment in above test)
```

Comment:
Just in case tests get moved around, could you name the test, ie Averager.Linear?